cmd/spiffe-aws-assume-role/cli/cli_test.go

```diff
@@ -0,0 +1,98 @@
+package cli
+
+import (
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/square/spiffe-aws-assume-role/internal/mocks"
+	"github.com/square/spiffe-aws-assume-role/internal/test"
+	"github.com/square/spiffe-aws-assume-role/pkg/credentials"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+var NoContext *CliContext = nil
+
+func TestParsesSessionDuration(t *testing.T) {
+	args := []string{
+		"credentials",
+		"--session-duration=PT5M",
+		// We only specify the following fields because they're required
+		"--audience=foo",
+		"--role-arn=bar",
+		"--spiffe-id=baz",
+	}
+	context, err := parse(args)
+	require.NoError(t, err)
+
+	cli := context.Model.Target.Interface().(CLI)
+	sessionDuration := cli.Credentials.SessionDuration
+	require.EqualValues(t, 5, sessionDuration.Minutes())
+}
+
+func TestDisplaysTopLevelHelp(t *testing.T) {
+	args := []string{"--help"}
+	require.NoError(t, Run(args, NoContext))
+}
+
+func TestDisplaysCredentialsHelp(t *testing.T) {
+	args := []string{"--help", "credentials"}
+	require.NoError(t, Run(args, NoContext))
+}
+
+func TestSetsCustomSessionDuration(t *testing.T) {
+	args := []string{
+		"credentials",
+		"--audience=foo",
+		"--role-arn=arn:aws:iam:123456789012:role/foo",
+		"--session-duration=PT5M",
+		"--spiffe-id=spiffe://foo",
+		"--workload-socket=tcp://127.0.0.1:8080",
+	}
+
+	jwtSource := mocks.JWTSource{}
+	defer jwtSource.AssertExpectations(t)
+	jwtSource.
+		On("FetchToken", mock.Anything).
+		Return("token", nil)
+
+	stsClient := mocks.STSAPI{}
+	defer stsClient.AssertExpectations(t)
+
+	expiration := time.Now()
+	emptyString := ""
+
+	stsCredentials := sts.Credentials{
+		AccessKeyId:     &emptyString,
+		Expiration:      &expiration,
+		SecretAccessKey: &emptyString,
+		SessionToken:    &emptyString,
+	}
```

Comment:
Credentials requires pointers and since you can't take the address of a literal we have to introduce some useless locals (and we can't just use nil since these are dereferenced in the business logic).

Reply:
you can use `aws.String("")` here. there's also `aws.StringValue` to go the other way

Reply:
also `aws.Time`

Reply:
Ooh fancy, I'll make that change.